File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.java
import java.util.*;
class Solution {
    public String solution(String number, int k) {
        String answer = "";
        StringBuilder sb = new StringBuilder();
        int index = 0;
        for(int i = 0; i < number.length()-k; i++) {
            int max = 0;
            for(int j = index; j <= i + k; j++) {
                int num = number.charAt(j) - '0';
                if(max < num) {
                    max = num;
                    index = j + 1;
                }
            }
            sb.append(String.valueOf(max));
        }
        return sb.toString();
    }
}

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
    var answer = '';
    const arr = [];
    for (let i = 0; i < number.length; i++) {
        while (k > 0 && arr[arr.length-1] < number[i]) {
            arr.pop();
            k--;
        }
        arr.push(number[i]);
    }
    
    answer = arr.slice(0,arr.length - k).join('');
    return answer;
}